core/system/src/pool_queue.c

```diff
@@ -361,10 +361,11 @@ static uvisor_pool_slot_t find_first(uvisor_pool_queue_t * pool_queue,
 {
     uvisor_pool_slot_t slot;
     uvisor_pool_t * pool = UVISOR_AUTO_NS_ALIAS(pool_queue->pool);
+    uvisor_pool_slot_t iterated = 0;
 
     /* Walk the queue, looking for the first slot that matches the query. */
     slot = pool_queue->head;
-    while (slot != UVISOR_POOL_SLOT_INVALID)
+    while (slot != UVISOR_POOL_SLOT_INVALID && iterated <= pool->num)
```

Comment:
Turns out `pool->num` is not a `uint32_t` or `size_t` as I had misremembered, so switching to `UVISOR_POOL_MAX_VALID` doesn't make too much of a difference.